example/linux/main.c

```diff
@@ -0,0 +1,89 @@
+
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdio.h>
+
+// to achieve more identical behaviour with mbed device you can active ncurses
+//#define EXAMPLE_USE_NCURSES 1
+
+#ifndef EXAMPLE_USE_NCURSES
+#define EXAMPLE_USE_NCURSES 0
+#endif
+#if EXAMPLE_USE_NCURSES == 1
+#include <ncurses.h>
+#endif
+#include "mbed-trace/mbed_trace.h"
+#include "ns_cmdline.h"
+
+#ifndef CTRL
+#define CTRL(c) ((c) & 037)
+#endif
+
+// dummy command with some option
+static int cmd_dummy(int argc, char *argv[]) {
+  if( cmd_has_option(argc, argv, "o") ) {
+    cmd_printf("This is o option\r\n");
+  } else {
+    return CMDLINE_RETCODE_INVALID_PARAMETERS;
+  }
+  return CMDLINE_RETCODE_SUCCESS;
+}
+volatile bool running = true;
+static int cmd_exit(int argc, char *argv[]) {
+  cmd_printf("Exiting cli.");
```

Comment:
seems this print didn't get printed when cmd is exit.

Reply:
fid you try with ncurses or without? most probably ncurses exit causes that..

Reply:
without `#define EXAMPLE_USE_NCURSES 0`
when `#define EXAMPLE_USE_NCURSES 1`, `./cli`, this `tr_info("write 'help' and press ENTER");` get printed after `exit`

Reply:
removed print since it's unnecessary